### PR TITLE
Add the possibility to switch off spdlog.

### PR DESCRIPTION
**Pull request by phaazon**
Relates to LLC-102.

### CODE REVIEW

File: core/src/debug/logger.cpp [hunk 1]
```diff
@@ -31,26 +31,42 @@
...
-            auto logger = spdlog::create(name, {logPrinterSink, rotatingSink});
-            spdlog::drop(name);
-            logger->set_level(spdlog::level::trace);
-            logger->flush_on(spdlog::level::trace);
-            logger->set_pattern("%Y-%m-%dT%XZ%z %L: %v");
-            return logger;
+
+            if (disabled) {
+                auto logger = spdlog::create<spdlog::sinks::null_sink_st>(name);
+                spdlog::drop(name);
```
Comment (teams2ua) [inline core/src/debug/logger.cpp]: As I understand from the doc https://github.com/gabime/spdlog/wiki/5.-Logger-registry 

> "The "drop()" function can be used to remove a logger from the registry.
> If no other shared_ptr pointing to the logger exists, the logger will be closed and all its resources will be freed."

Will be the `logger` still alive in the next line?

Comment (phaazon) [reply core/src/debug/logger.cpp]: I don’t really know why we do that since that line was already there prior to my PR. :confused:

Comment (teams2ua) [reply core/src/debug/logger.cpp]: Ah I see, somebody else should call `spdlog::register_logger(some_logger);` latter, strange we have this line here. OK

**Comment by phaazon** [comment]
I’m still leaving it tagged until I have added some kind of a test to ensure no file gets generated.

**Comment by phaazon** [comment]
I shall merge this as I tested it correctly. I’ll make another PR with the IDL commits.

**Pull request merged**